networking/src/Mockable/Concurrent.hs

```diff
@@ -227,6 +231,57 @@ forConcurrently
     -> m (f t)
 forConcurrently = flip mapConcurrently
 
+----------------------------------------------------------------------------
+-- Low-level Async mock and helper functions
+----------------------------------------------------------------------------
+
+data LowLevelAsync m t where
+    Async :: m t -> LowLevelAsync m (Promise m t)
+    Link :: Promise m t -> LowLevelAsync m ()
+    Wait :: Promise m t -> LowLevelAsync m t
+    WaitAny :: [Promise m t] -> LowLevelAsync m (Promise m t, t)
+    CancelWith :: Exception e => Promise m t -> e -> LowLevelAsync m ()
```

Comment:
Why `LowLevelAsync`? Was having these in `Async` considered dangerous? If so, why?

Reply:
This is one of recommendations from FP-Complete. See https://docs.google.com/document/d/1ueuoII0UIkHbj_8eozJ_3InkLTRc0wTSBvu7g6RcrOk/edit#heading=h.ar36nz4qljge.

Usually these functions are not needed, higher-level abstractions are enough. We don't want to completely prohibit usage of these functions, but we want to make it clearer in which places they are used.